libraries/Bluefruit52Lib/src/services/BLEDis.h

```diff
@@ -46,24 +46,56 @@ class BLEDis : public BLEService
   protected:
     union {
       struct {
+        const char * _system_id;
         const char * _model;
         const char * _serial;
         const char * _firmware_rev;
         const char * _hardware_rev;
         const char * _software_rev;
         const char * _manufacturer;
+        const char * _reg_cert_list;
+        const char * _pnp_id;
       };
 
-      const char * _strarr[6];
+    const char * _strarr[9];
+    };
+
+    union {
+      struct {
+        uint8_t _system_id_length;
+        uint8_t _model_length;
+        uint8_t _serial_length;
+        uint8_t _firmware_rev_length;
+        uint8_t _hardware_rev_length;
+        uint8_t _software_rev_length;
+        uint8_t _manufacturer_length;
+        uint8_t _reg_cert_list_length;
+        uint8_t _pnp_id_length;
+      };
+      const uint8_t _strarr_length[9];
     };
 
   public:
     BLEDis(void);
 
     void setModel(const char* model);
+    void setModel(const char* model,uint8_t length);
```

Comment:
for using this API with length, I would suggest to 
1. change the type of 1st parameter to `const uint8_t*` e.g `setModel(const uint8_t model, uint8_t length)`
2. also please implement the `const char*` as inline function as well.

Reply:
1. Yes, you are right. Then inside function it must be "_model = (const char*)model;"
2. Sorry, I don't know what to change, or do you mean "_model = (const char*)model;"